src/NSwag.MSBuild.CodeGeneration/NSwag.MSBuild.CodeGeneration.targets

```diff
@@ -0,0 +1,94 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <PropertyGroup>
+    <_NSwagcommand>$(NSwagExe)</_NSwagcommand>
+    <_NSwagcommand Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NSwagExe_Core20)</_NSwagcommand>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <CurrentServiceProjectReference>
+      <GenerateNSwagDocumentOptions />
+      <Configuration />
+      <Platform />
+    </CurrentServiceProjectReference>
+    <CurrentServiceFileReference>
+      <GenerateNSwagCSharpOptions />
+      <GenerateNSwagTypeScriptOptions />
+    </CurrentServiceFileReference>
+  </ItemDefinitionGroup>
+
+  <!-- ServiceProjectReference support -->
+
+  <Target Name="GenerateNSwagDocument">
+    <ItemGroup>
+      <!-- @(CurrentServiceProjectReference) item group will never contain more than one item. -->
+      <CurrentServiceProjectReference Update="@(CurrentServiceProjectReference)">
+        <Command>$(NSwagExe)</Command>
+        <Command Condition="'%(Platform)' == 'x86'">$(NSwagExe_x86)</Command>
+        <Command Condition="'%(TargetFramework)' == 'netcoreapp1.0'">$(NSwagExe_Core10)</Command>
+        <Command Condition="'%(TargetFramework)' == 'netcoreapp1.1'">$(NSwagExe_Core11)</Command>
+        <Command Condition="'%(TargetFramework)' == 'netcoreapp2.0'">$(NSwagExe_Core20)</Command>
+        <Command Condition="'%(TargetFramework)' == 'netcoreapp2.1'">$(NSwagExe_Core21)</Command>
```

Comment:
Since `netcoreapp2.2` is available, I should add a case for that too (using `$(NSwagExe_Core21)` for now).

Reply:
Added new issue: https://github.com/RSuter/NSwag/issues/1663
For now, just use 2.1

Reply:
Do you mean leave this block as-is or add `<Command Condition="'%(TargetFramework)' == 'netcoreapp2.2'">$(NSwagExe_Core21)</Command>` i.e. use 2.1 for a 2.2 web site?

Either is :ok: with me.

Reply:
I think support will be better if you'd add 2.2 and use NSwagExe_Core21. Is this even working?

Reply:
Done and yes it works